stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -693,3 +693,150 @@ func.func @pad_with_negative_inferred_bounds(%arg0: tensor<3x?x?xf16, #stablehlo
   %1 = "hlo_test_infer.get_return_types"(%0) : (tensor<*xf16>) -> tensor<*xindex>
   func.return %1 : tensor<*xindex>
 }
+
+// -----
+
+// Alse see this in Base.cpp
+// Inference rules to concat dimensions with bounds (lhs/rhs are commutative):
+//       Dim of lhs     Dim of rhs      Infer
+//  c0:  X              Y               X+Y
+//  c1:  X              ?               ?
+//  c2:  X              ?, B            ?, X+B
+//  c3:  ?              ?               ?
+//  c4:  ?              ?, B            ?
+//  c5:  ?, B           ?, C            ?, B+C
```

Comment:
Also, let's simplify ranked tests by only using one dimension, i.e. instead of `5x1xi32` and `dimension = 1`, let's use `1xi32` and `dimension = 0` etc.

Reply:
That is done on purpose for this part of code in `inferConcatenateOp()`:
```
if (dim == dimension) {
  inferredDimAndBound = inferConcatenatedDimAndBound(
      leftSize, rightSize, leftBound, rightBound);
} else {
  auto inferredDimAndBoundOrErr = inferMergedDimAndBound(
      location, dim, leftSize, rightSize, leftBound, rightBound);
  if (failed(inferredDimAndBoundOrErr)) return failure();
  inferredDimAndBound = *inferredDimAndBoundOrErr;
}
```
Removing that `5` dim will weaken the tests for no covering the `else` part above.

Reply:
Got it, let's keep the tests as is then.